core-web/src/main/java/org/silverpeas/core/web/session/SessionManager.java

```diff
@@ -540,12 +536,16 @@ public SessionInfo openSession(User user, HttpServletRequest request) {
       // In case of error, simply taking the value from the servlet request
       anIP = requestRemoteHost;
     }
+
+    HttpSession session = request.getSession();
+    HTTPSessionInfo si = new HTTPSessionInfo(session, anIP, user);
     try {
-      HttpSession session = request.getSession();
-      si = new HTTPSessionInfo(session, anIP, user);
       openSession(si);
-      userDataSessions.put(si.getSessionId(), si);
-      defaultServerEventNotifier.notify(UserSessionServerEvent.anOpeningOneFor(si));
+      if (!si.isAnonymous()) {
+        // only non-anonymous session requires to be monitored by the manager
+        userDataSessions.put(si.getSessionId(), si);
+        defaultServerEventNotifier.notify(UserSessionServerEvent.anOpeningOneFor(si));
+      }
```

Comment:
The content of the "if" statement could be moved to the private method openSession(SessionInfo).